api/v1beta1/kogitoinfra_types.go

```diff
@@ -103,13 +103,14 @@ func (k *KogitoInfraStatus) GetRuntimeProperties() api.RuntimePropertiesMap {
 	return runtimeProperties
 }
 
-// SetRuntimeProperties ...
-func (k *KogitoInfraStatus) SetRuntimeProperties(runtimeProperties api.RuntimePropertiesMap) {
-	newRuntimeProperties := make(map[api.RuntimeType]*RuntimeProperties, len(runtimeProperties))
-	for key, value := range runtimeProperties {
-		newRuntimeProperties[key] = value.(*RuntimeProperties)
+// AddRuntimeProperties ...
+func (k *KogitoInfraStatus) AddRuntimeProperties(runtimeType api.RuntimeType, runtimeProperties api.RuntimePropertiesInterface) {
+	rp := k.RuntimeProperties
+	if rp == nil {
+		rp = make(map[api.RuntimeType]RuntimeProperties)
 	}
-	k.RuntimeProperties = newRuntimeProperties
+	rp[runtimeType] = runtimeProperties.(RuntimeProperties)
+	k.RuntimeProperties = rp
```

Comment:
According to test it would be enough to have:
```
	if k.RuntimeProperties == nil {
		k.RuntimeProperties = make(map[api.RuntimeType]RuntimeProperties)
	}
	k.RuntimeProperties[runtimeType] = runtimeProperties.(RuntimeProperties)
```

Reply:
done